### PR TITLE
Remove feature gate for ILB Custom Subnet.

### PR DESCRIPTION
**Pull request by prameshj**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:
This PR removes the feature gate for ILB Custom Subnet. The annotation to specify a custom subnet was introduced in 1.17 under a feature gate. Targeting this change for 1.20.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by prameshj** [comment]
/hold
/sig network

This will be targeted for 1.19, just a placeholder for now.

**Comment by stiller-leser** [comment]
Did it ever make its way into 1.19?

**Comment by prameshj** [comment]
Thanks! didn't realise this got auto-closed.
/reopen
/remove-lifecycle rotten

File: staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go [hunk 1]
```diff
@@ -124,20 +118,12 @@ func (g *Cloud) ensureInternalLoadBalancer(clusterName, clusterID string, svc *v
...
-			subnetworkURL = gceSubnetworkURL("", g.networkProjectID, g.region, options.SubnetName)
-		}
-	} else {
-		// TODO(84885) remove this once ILBCustomSubnet goes beta.
-		if existingFwdRule != nil && existingFwdRule.Subnetwork != "" {
-			// If the ILB already exists, continue using the subnet that it's already using.
-			// This is to support existing ILBs that were setup using the wrong subnet - https://github.com/kubernetes/kubernetes/pull/57861
-			subnetworkURL = existingFwdRule.Subnetwork
-		}
+	// If this feature is enabled, changes to subnet annotation will be picked up and reflected in the forwarding rule.
```
Comment (MrHohn) [inline staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go:121]: Sorry dumb question - did the annotation graduate to beta/ga or is this removing it completely? Is this comment still valid?

Comment (prameshj) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go:121]: we graduated this support to GA with the annotation intact. So now cleaning up the annotation. Good point, I will remove the comment here.

Comment (MrHohn) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go:121]: Ah got it, thanks for the explanation.

Comment (prameshj) [reply staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go]: fixed the comment @MrHohn

**Comment by cici37** [comment]
There is an known issue for test infra: kubernetes/test-infra#19910

**Comment by cici37** [comment]
/lgtm
This PR has to wait for 1.21 hence would be affected by https://github.com/kubernetes/kubernetes/pull/97044

**Comment by cheftako** [comment]
> /lgtm
> This PR has to wait for 1.21 hence would be affected by #97044

Happy to get @andrewsykim take but this seems to be cleaning up a flag rather than adding a feature. As it touches the GCP Cloud Provider and I don't want to be biased I'll let Andrew make the call.

**Pull request merged**